src/org/infinity/search/StringReferenceSearcher.java

```diff
@@ -25,16 +24,36 @@
 import org.infinity.resource.sav.SavResource;
 import org.infinity.resource.text.PlainTextResource;
 
+/** Performs search of the specified string reference in other resources. */
 public final class StringReferenceSearcher extends AbstractReferenceSearcher
 {
-  private static final Pattern NUMBERPATTERN = Pattern.compile("\\d+", Pattern.DOTALL);
+  /**
+   * Regular expression, used to the find string references in the
+   * {@link PlainTextResource text resources}.
+   * Expression consists from three parts:
+   * <ul>
+   * <li>lookbehind for a dot that filters fractional part of the floating-point value</li>
+   * <li>integer part itself surrounded with word boundaries to filter things like 2 in {@code 2DA}</li>
+   * <li>lookahead for a dot that filter integer part of the floating-point value</li>
+   * </ul>
+   */
+  public static final Pattern NUMBER_PATTERN = Pattern.compile("(?<!\\.)" + "\\b\\d+\\b" + "(?!\\.)", Pattern.DOTALL);
```

Comment:
Now the same pattern used in all places where string reference is searched: `StringUseChecker`, `StrrefIndexChecker` and this. Also now it is more accurate